app/src/main/java/ch/hikemate/app/ui/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,149 @@
+ * @param profileViewModel The profile view model.
+ */
+@Composable
+fun ProfileScreen(
+    navigationActions: NavigationActions,
+    profileViewModel: ProfileViewModel = viewModel(factory = ProfileViewModel.Factory)
+) {
+  val context = LocalContext.current
+
+  // TODO: show an error if the profile is null. For now display it for test purposes
+  val profileState = profileViewModel.profile.collectAsState()
+
+  val profile: Profile =
+      profileState.value
+          ?: Profile(
+              "custom-id",
+              "John Doe",
+              "john.doe@gmail.com",
+              HikingLevel.INTERMEDIATE,
+              Timestamp.now())
+
+  SideBarNavigation(
+      onTabSelect = { navigationActions.navigateTo(it) },
+      tabList = LIST_TOP_LEVEL_DESTINATIONS,
+      selectedItem = Route.PROFILE) { p ->
+        Column(
+            modifier =
+                Modifier.testTag(Screen.PROFILE)
+                    .padding(p)
+                    .padding(
+                        // Add padding to the sidebar padding
+                        start = 16.dp,
+                        end = 16.dp,
+                        top = 16.dp,
+                    ),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
+              Text(
+                  context.getString(R.string.profile_screen_title),
+                  style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 32.sp),
+                  modifier = Modifier.testTag(ProfileScreen.TEST_TAG_TITLE))
+
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_name_label),
+                  profile.name,
+                  Modifier.testTag(ProfileScreen.TEST_TAG_NAME))
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_email_label),
+                  profile.email,
+                  Modifier.testTag(ProfileScreen.TEST_TAG_EMAIL))
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_hiking_level_label),
+                  buildString {
+                    append(context.getString(R.string.profile_screen_hiking_level_info_prefix))
+                    append(" ")
+                    append(
+                        when (profile.hikingLevel) {
+                          HikingLevel.BEGINNER ->
+                              context.getString(R.string.profile_screen_hiking_level_info_beginner)
+                          HikingLevel.INTERMEDIATE ->
+                              context.getString(
+                                  R.string.profile_screen_hiking_level_info_intermediate)
+                          HikingLevel.EXPERT ->
+                              context.getString(R.string.profile_screen_hiking_level_info_expert)
+                        })
+                    append(" ")
+                    append(context.getString(R.string.profile_screen_hiking_level_info_suffix))
+                  },
```

Comment:
Nitpick: Maybe it would be better for clarity to build the string outside the composable?

Reply:
Or just put the whole string in string.xml. Would also be easier for (future) translations